### PR TITLE
Circulating Supply

### PR DESCRIPTION
**Pull request by jesus-nuco**
This PR has changes related to the ticket DO-291.

Changes:
Removed the hardcoded records for total supply by an automatic process as follow
`Total circulating supply= (Initial network allocation balance)+ (Sum of all block rewards) - (Balance of the burn address 0x00)`

### CODE REVIEW

File: aion-api/src/main/java/com/aion/dashboard/controllers/Dashboardv2.java [hunk 1]
```diff
@@ -667,4 +651,12 @@ boolean isNotEmpty(String str){
             blockNumber.map(statisticsService::transactionStats).orElseGet(statisticsService::transactionStats)
         ));
     }
+
+    @GetMapping(value = "/circulatingSupply")
+    public ResponseEntity<Result<CirculatingSupplyDTO>> getCirculatingSupply(){
+        return packageResponse(CirculatingSupplyMapper
+                .getMapper()
+                .makeResult(circulatingSupplyJPARepository.findAll().get(0)));
```
Comment (qoire) [inline aion-api/src/main/java/com/aion/dashboard/controllers/Dashboardv2.java:659]: I'm not sure about how efficienct this is under the hood (whether the query is lazy or not) but it seems like you could make this more efficient by indexing on timestamp, and querying by the latest entry from the last day or so. Also is there caching for this endpoint?

File: aion-api/src/main/java/com/aion/dashboard/services/ThirdPartyService.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public String getCirculatingSupply() throws IOException {
             List<CirculatingSupply> circulatingSupplies = circulatingSupply.findAll();//this fails on test nets
             TreeMap<Instant, BigInteger> temp= new TreeMap<>(Instant::compareTo);
             for (CirculatingSupply supply : circulatingSupplies) {//read the table and store the contents in memory
-                temp.put(supply.getStartDate().toInstant(), supply.getSupply().toBigInteger());
+                temp.put(supply.getTimestamp().toInstant(), supply.getSupply().toBigInteger());
```
Comment (qoire) [inline aion-api/src/main/java/com/aion/dashboard/services/ThirdPartyService.java:36]: Not sure I fully follow the intention behind this endpoint, when does this cache get invalidated? (It seems like circulatingSupplyMap will get instantiated on the first call, then never be null again correct?)

**Review by qoire — CHANGES_REQUESTED** [review]

File: aion-api/src/main/java/com/aion/dashboard/services/ThirdPartyService.java (hunk 2 above)
Comment (jesus-nuco) [reply aion-api/src/main/java/com/aion/dashboard/services/ThirdPartyService.java]: I removed this class, with the new implementation it got deprecated

File: aion-api/src/main/java/com/aion/dashboard/controllers/Dashboardv2.java (hunk 1 above)
Comment (jesus-nuco) [reply aion-api/src/main/java/com/aion/dashboard/controllers/Dashboardv2.java]: Index, cache, and limiting query  was added

File: aion-api/src/main/java/com/aion/dashboard/controllers/Dashboard.java [hunk 3]
```diff
@@ -501,16 +503,16 @@ public String viewDashboard() {
...
-
-    // Third Party APIs
+    @Cacheable(CacheConfig.CIRCULATING_SUPPLY)
     @GetMapping(value = "/getCirculatingSupply")
     public String getCirculatingSupply() {
         try {
-            return thirdPartyService.getCirculatingSupply();
+             return circulatingSupplyJPARepository
+                     .findByOrderByTimestampDesc(PageRequest.of(0, 1)).getContent().get(0).
+                             getSupply().setScale(0, BigDecimal.ROUND_HALF_EVEN).toString();
```
Comment (qoire) [inline aion-api/src/main/java/com/aion/dashboard/controllers/Dashboard.java:512]: Recommended to use `RoundingMode.HALF_EVEN` instead

**Review by qoire — CHANGES_REQUESTED** [review]

**Comment by qoire** [comment]
Just want to confirm: the endpoint we're updating originally only returned a string containing an _integer_ value, so no decimals. In which case LGTM.

File: aion-api/src/main/java/com/aion/dashboard/controllers/Dashboard.java (hunk 3 above)
Comment (qoire) [reply aion-api/src/main/java/com/aion/dashboard/controllers/Dashboard.java:512]: FYI, this is on JDK 11, if the server is not using that version of JDK then my bad, go with what is appropriate.

Comment (jesus-nuco) [reply aion-api/src/main/java/com/aion/dashboard/controllers/Dashboard.java:512]: yes, we are using JDK11, so I will change

**Review by qoire — APPROVED** [review]

**Pull request closed without merging**